WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -530,17 +530,27 @@ FROM INFORMATION_SCHEMA.ROUTINES
 
                 // Rollback transaction if started
                 await databaseConnection.RollbackTransactionAsync(false);
+                logger.LogError(exception, $"Failed to create the branch '{settings.DatabaseName}'.");
 
                 // Save the error in the queue and set the finished on datetime to now.
                 await FinishBranchActionAsync(queueId, dataRowWithSettings, branchQueue, configurationServiceName, databaseConnection, wiserItemsService, taskAlertsService, String.IsNullOrWhiteSpace(error) ? new JArray() : new JArray(error), stopwatch, startDate, branchQueue.CreatedBranchTemplateId, CreateBranchSubject, CreateBranchTemplate);
 
                 // Drop the new database it something went wrong, so that we can start over again later.
                 // We can safely do this, because this method will return an error if the database already exists,
                 // so we can be sure that this database was created here and we can drop it again it something went wrong.
-                if (await databaseHelpersService.DatabaseExistsAsync(branchDatabase))
+                try
                 {
-                    await databaseHelpersService.DropDatabaseAsync(branchDatabase);
+                    if (await databaseHelpersService.DatabaseExistsAsync(branchDatabase))
+                    {
+                        await databaseHelpersService.DropDatabaseAsync(branchDatabase);
+                    }
+                }
+                catch (Exception innerException)
+                {
+                    logger.LogError(innerException, "Failed to drop new branch database, after getting an error while trying to fill it with data.");
```

Comment:
Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.

Reply:
Done